src/resolve/resolved-manager.c

```diff
@@ -378,11 +378,19 @@ static int on_clock_change(sd_event_source *source, int fd, uint32_t revents, vo
          *
          * (Also, this is triggered after system suspend, which is also a good reason to drop caches, since
          * we might be connected to a different network now without this being visible in a dropped link
-         * carrier or so.) */
-
-        log_info("Clock change detected. Flushing caches.");
-        manager_flush_caches(m, LOG_DEBUG /* downgrade the functions own log message, since we already logged here at LOG_INFO level */);
-
+         * carrier or so.) 
+         * 
+         * That being said, old data isn't neccessarily invalid; so if we can't reach upstream DNS immediately,
+         * provide stale data if configured to.  */
+        
+        if(m->stale_retention_usec == 0){
+                log_info("Clock change detected. Flushing caches.");
+                manager_flush_caches(m, LOG_DEBUG /* downgrade the functions own log message, since we already logged here at LOG_INFO level */);
+        } else {
+                log_info("Clock change detected. Marking caches as stale.");
+                manager_force_caches_stale(m, LOG_DEBUG /* downgrade the functions own log message, since we already logged here at LOG_INFO level */);
```

Comment:
I cannot follow why this is called here, rather than when `PrepareForSleep` signal is received, like systemd-networkd does.

Reply:
This is how the existing code works; modifying that is out of scope for this PR.